core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -283,14 +283,15 @@ class GameInfo : IsPartOfGameInfoSerialization, HasGameInfoSerializationVersion
                     break
                 }
             }
-            switchTurn()
+            endTurn()
         }
         if (turns == UncivGame.Current.simulateUntilTurnForDebug)
             UncivGame.Current.simulateUntilTurnForDebug = 0
 
         currentTurnStartTime = System.currentTimeMillis()
         currentPlayer = thisPlayer.civName
         currentPlayerCiv = getCivilization(currentPlayer)
+        thisPlayer.startTurn()
```

Comment:
Need to move this function down to below the point where the `currentPlayer` switches to the next human player so that `gameInfo.isUsersTurn()` is looking at the next human rather than the human player who just clicked "Next Turn". This prevents the game from automating the production of the next human player if the one who just ended their turn has the automate production setting checked.